docs/src/treasury.md

```diff
@@ -1 +1,257 @@
+    };
+
+    assert(
+        pooled_token.balanceOf(get_contract_address()) >= amount,
+        Errors::INSUFFICIENT_POOLED_TOKEN
+    );
+    pooled_token.approve(self.amm_address.read(), amount);
+
+    carm_AMM.deposit_liquidity(
+        pooled_token_addr, quote_token_address, base_token_address, option_type, amount
+    );
+    self.emit(LiquidityProvided {
+        quote_token_address, base_token_address, option_type, amount
+    });
+}
+```
+
+#### `withdraw_liquidity`
+
+Withdraws liquidity from the Carmine AMM. Only the contract owner can execute this function.
+
+```cairo
+fn withdraw_liquidity(
+    ref self: ContractState,
+    pooled_token_addr: ContractAddress,
+    quote_token_address: ContractAddress,
+    base_token_address: ContractAddress,
+    option_type: OptionType,
+    lp_token_amount: u256
+) {
+    self.ownable.assert_only_owner();
+    let carm_AMM: IAMMDispatcher = IAMMDispatcher {
+        contract_address: self.amm_address.read()
+    };
+
+    let lp_token_addr = carm_AMM.get_lptoken_address_for_given_option(
+        quote_token_address, base_token_address, option_type
+    );
+    let lp_token: IERC20Dispatcher = IERC20Dispatcher { contract_address: lp_token_addr };
+    assert(
+        lp_token.balanceOf(get_contract_address()) >= lp_token_amount,
+        Errors::INSUFFICIENT_LP_TOKENS
+    );
+
+    carm_AMM.withdraw_liquidity(
+        pooled_token_addr,
+        quote_token_address,
+        base_token_address,
+        option_type,
+        lp_token_amount
+    );
+    self.emit(LiquidityWithdrawn {
+        quote_token_address, base_token_address, option_type, lp_token_amount
+    });
+}
+```
+
+#### `get_amm_address`
+
+Returns the current address of the AMM contract.
+
+```cairo
+fn get_amm_address(self: @ContractState) -> ContractAddress {
+    self.amm_address.read()
+}
+```
+
+## Events
+
+The `Treasury` component emits several events to notify about significant actions:
+- `TokenSent`: Emitted when tokens are sent to an address.
+- `AMMAddressUpdated`: Emitted when the AMM address is updated.
+- `LiquidityProvided`: Emitted when liquidity is provided to the Carmine AMM.
+- `LiquidityWithdrawn`: Emitted when liquidity is withdrawn from the Carmine AMM.
+
+## Error Handling
+
+The `Treasury` component defines several error messages for assertion checks:
+- `INSUFFICIENT_FUNDS`: Insufficient token balance for the transfer.
+- `INSUFFICIENT_POOLED_TOKEN`: Insufficient pooled token balance for providing liquidity.
+- `INSUFFICIENT_LP_TOKENS`: Insufficient LP token balance for withdrawing liquidity.
+- `ADDRESS_ZERO_GOVERNANCE`: Governance address is zero.
+- `ADDRESS_ZERO_AMM`: AMM address is zero.
+- `ADDRESS_ALREADY_CHANGED`: New AMM address is the same as the current one.
+
+## Customization
+
+The `Treasury` component can be customized by:
+- Managing the ownership of the contract through OpenZeppelin's `OwnableComponent`.
+- Upgrading the contract using OpenZeppelin's `UpgradeableComponent`.
```

Comment:
What question does this answer, for whom is this written? How are you _customizing_ the contract by managing ownership??